lib/internal/modules/esm/loader.js

```diff
@@ -156,19 +151,20 @@ class ESMLoader {
    */
   translators = translators;
 
-  constructor() {
-    if (getOptionValue('--experimental-loader')) {
+  constructor({ isInternal = false } = {}) {
```

Comment:
```suggestion
  constructor({ isInternal = false } = ObjectCreate(null)) {
```
but I don't think we need to go the object route, it's an internal constructor anyway, we'll make it an object if we ever add other parameters:
```suggestion
  constructor(isInternal = false) {
```

Reply:
When do we use a null-prototype object and when don't we?

Reply:
My reason for using an object is it is self-documenting:

```js
new ESMLoader({ isInternal: true }) // I immediately see what this is &
                                    // have a good idea what it does
new ESMLoader(true) // No idea 🤷‍♂️
```

god help you if you have to instantiate a `ModuleJob`

Reply:
Ahh, okay, so just when there's a danger of pollution from user-land. Thanks!

Reply:
I thought we were removing `isInternal`, though?

Reply:
We did but now it's back because a bajillion tests were now failing, and because I figured out how to get it to work for skipping the warnings. This also reduces a lot of noise in test runs.

I replaced that `let` workaround you added. Was gonna tag you as soon as the current test run completes so I can show you the difference.

Reply:
RE cutting down on noise: in the linux run, it's 7k lines shorter now